File: leetcode/go/1566.detect-pattern-of-length-m-repeated-k-or-more-times.go
func containsPattern(arr []int, m int, k int) bool {
    count := 0
    for i := m ; i < len(arr) ; i++ {
        if arr[i] == arr[i-m] {
            count++
        } else {
            count = 0
        }
        if count == m * k - m {
            return true
        }
    }
    return false
}

File: leetcode/rust-leetcode/src/1566.detect-pattern-of-length-m-repeated-k-or-more-times.rs
impl Solution {
    pub fn contains_pattern(arr: Vec<i32>, m: i32, k: i32) -> bool {
        let m = m as usize;
        let k = k as usize;
        let mut count = 0;
        for i in m..arr.len() {
            count = if arr[i] == arr[i-m] {count+1} else {0};
            if count == m * k - m {
                return true;
            }
        }
        false
    }
}

